### PR TITLE
feat: fix server exit or restart signal

### PR DESCRIPTION
**Pull request by daheige**
**Please provide issue(s) of this PR:**
Fixes https://github.com/polarismesh/polaris/issues/769
fix server exit or restart signal

### CODE REVIEW

File: bootstrap/run_linux.go [hunk 1]
```diff
@@ -31,32 +31,28 @@ var linuxSignals = []os.Signal{
...
-					log.Infof("restart servers success: %+v", s)
-					continue
-				}
+
+	select {
+	case s := <-ch:
+		// restart信号
+		if s.(syscall.Signal) == syscall.SIGUSR1 {
+			// 注意：重启失败，退出程序
+			if err := RestartServers(errCh); err != nil {
```
Comment (andrewshan) [inline bootstrap/run_linux.go:46]: 这里去掉循环的话，重启时候就直接退出了，达不到原地重启的效果？

Comment (daheige) [reply bootstrap/run_linux.go]: This has been fixed

File: bootstrap/run_darwin.go [hunk 2]
```diff
@@ -26,34 +26,46 @@ import (
...
-			return
-		case err := <-errCh:
-			log.Errorf("catch api server err: %s", err.Error())
-			return
+				log.Infof("catch signal(%s), stop servers", s.String())
+				return
+			case err := <-errCh:
+				log.Errorf("catch api server err: %s", err.Error())
+				return
+			default:
```
Comment (srstack) [inline bootstrap/run_darwin.go]: please delete the default case, it will  generate unnecessary cost

File: bootstrap/run_linux.go [hunk 3]
```diff
@@ -26,33 +26,38 @@ import (
...
 					return
-				} else {
-					log.Infof("restart servers success: %+v", s)
-					continue
 				}
+
+				log.Infof("restart servers success: %s", s.String())
+				// 重启成功，就需要监听信号量然后执行相应的操作
+				signal.Notify(ch, linuxSignals...)
+				break label
```
Comment (chuntaojun) [inline bootstrap/run_linux.go:57]: 和continue 的区别？如果不是要从内层for 跳到外层for，或者跳到另外的逻辑？是不是不需要这个break label

**Comment by daheige** [comment]
There is a difference between this and continue. The break label continues to monitor semaphores after the service is restarted.


------------------ Original ------------------
From: liaochuntao ***@***.***&gt;
Date: Wed,Oct 26,2022 8:03 AM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: fix server exit or restart signal (PR #768)





 
@chuntaojun commented on this pull request.
 
 
In bootstrap/run_linux.go:
 &gt;  } + +log.Infof("restart servers success: %s", s.String()) +// 重启成功，就需要监听信号量然后执行相应的操作 +signal.Notify(ch, linuxSignals...) +break label  
和continue 的区别？如果不是要从内层for 跳到外层for，或者跳到另外的逻辑？是不是不需要这个break label

**Comment by daheige** [comment]
This is to jump out of the internal for and continue to monitor semaphores.


------------------ Original ------------------
From: liaochuntao ***@***.***&gt;
Date: Wed,Oct 26,2022 8:03 AM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: fix server exit or restart signal (PR #768)





 
@chuntaojun commented on this pull request.
 
 
In bootstrap/run_linux.go:
 &gt;  } + +log.Infof("restart servers success: %s", s.String()) +// 重启成功，就需要监听信号量然后执行相应的操作 +signal.Notify(ch, linuxSignals...) +break label  
和continue 的区别？如果不是要从内层for 跳到外层for，或者跳到另外的逻辑？是不是不需要这个break label

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**